pkg/network/network_test.go

```diff
@@ -539,6 +539,22 @@ func TestRewriteHost(t *testing.T) {
 		t.Errorf("r.Header[%s] = %q	, want: %q", OriginalHostHeader, got, want)
 	}
 
+	// Do it again, but make sure that the ORIGINAL domain is still preserved.
+	r.Header.Set("Host", "hate.is")
+	RewriteHostIn(r)
+
+	if got, want := r.Host, ""; got != want {
+		t.Errorf("r.Host = %q, want: %q", got, want)
+	}
+
+	if got, want := r.Header.Get("Host"), ""; got != want {
+		t.Errorf("r.Header['Host'] = %q, want: %q", got, want)
+	}
+
+	if got, want := r.Header.Get(OriginalHostHeader), "love.is"; got != want {
+		t.Errorf("r.Header[%s] = %q	, want: %q", OriginalHostHeader, got, want)
```

Comment:
s/\t/ /. Same above in my code. 
/shrug if I know how it happened.

Reply:
Done